### ISSUE TITLE
Blurhash is not generated

### ISSUE THREAD

**Issue by oezkancodes** [Demonstrandum/strapi-blurhash-plugin#1]
* `@strapi/strapi` version: 4.16.2
* `strapi-blurhash-plugin` version: 0.2.1
* Plugin gets recognized by Strapi

Tried uploading JPEG, PNG, SVG & WebP. Nothing generated the blur hash. The field `blurhash` always stays `null`.

```json
"image": {
    "data": {
        "id": 34,
        "attributes": {
            "name": "example.webp",
            "alternativeText": null,
            "caption": null,
            "width": 700,
            "height": 700,
            "formats": {},
            "hash": "example_89a42363d6",
            "ext": ".webp",
            "mime": "image/webp",
            "size": 31.09,
            "url": "/uploads/example_89a42363d6.webp",
            "previewUrl": null,
            "provider": "local",
            "provider_metadata": null,
            "createdAt": "2024-01-11T13:39:37.754Z",
            "updatedAt": "2024-01-11T13:39:37.754Z",
            "blurhash": null
        }
    }
},
```

My `plugins.ts` config file:

```ts
export default ({ env }) => ({
  "strapi-blurhash-plugin": {
    enabled: true,
    config: {
      regenerateOnUpdate: true,
    },
  },
});

```

Am I missing something?

**Comment by pixelume** [comment]
Yep. I am experiencing the same issue.

**Comment by Dominic-Preap** [comment]
I got the same issue, turn out if using local upload we got INVALID URL error since the url will be `/uploads/image-name.png` but if using other upload provider such as s3, I can generate blurhash.

![image](https://github.com/Demonstrandum/strapi-blurhash-plugin/assets/14802170/fee1d8cf-912d-4fd8-9ca7-49ab3c4bdb32)


The code snippet below work only full URL get from `files` table `url` field, so local upload provider does not work, I suggest having base URL in `config/plugins`  in case using local upload.

```ts
const loadImage = (url: string): Promise<Buffer> => new Promise((resolve, reject) => {
    https.get(url, res => {
        const data = [];
        res.on('data', chunk => data.push(chunk))
           .on('error', e => reject(e))
           .on('end', () => resolve(Buffer.concat(data)));
    });
});
```

**Comment by pixelume** [comment]
> I got the same issue, turn out if using local upload we got INVALID URL error since the url will be `/uploads/image-name.png` but if using other upload provider such as s3, I can generate blurhash.
> 
> ![image](https://private-user-images.githubusercontent.com/14802170/298378207-fee1d8cf-912d-4fd8-9ca7-49ab3c4bdb32.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU4NTgxMDcsIm5iZiI6MTcwNTg1NzgwNywicGF0aCI6Ii8xNDgwMjE3MC8yOTgzNzgyMDctZmVlMWQ4Y2YtOTEyZC00ZmQ4LTljYTctNDlhYjNjNGJkYjMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIxVDE3MjMyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFlMDU2OTM2OWZkOTdjMjdiOTY1NjM5M2I2YTc3NjJhMmFjMGZhOGVjYTU0MDJlMmRjYjEzNGJiMjRmNGViNWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-hW2RvwYeaTLrZIkNjr3OS0B5RHvqEmJYuqDx7Gjbxc)
> 
> The code snippet below work only full URL get from `files` table `url` field, so local upload provider does not work, I suggest having base URL in `config/plugins` in case using local upload.
> 
> ```ts
> const loadImage = (url: string): Promise<Buffer> => new Promise((resolve, reject) => {
>     https.get(url, res => {
>         const data = [];
>         res.on('data', chunk => data.push(chunk))
>            .on('error', e => reject(e))
>            .on('end', () => resolve(Buffer.concat(data)));
>     });
> });
> ```

@Dominic-Preap Rockstar! Thanks so much it works.

**Comment by tennox** [comment]
@Dominic-Preap 
> The code snippet below

How to use this snippet?

**Comment by Demonstrandum** [comment]
Sorry for the late response.

I merged (5142bd8) a commit by @tennox and published it under version `v0.2.2`, which should hopefully resolve this issue.

If not, please let me know and reopen the issue.

**Comment by Demonstrandum** [comment]
Updated fix was released (`v0.2.3`).
